docs/api-reference/layers/geojson-layer.md

```diff
@@ -354,6 +354,156 @@ The GeoJsonLayer renders the following sublayers:
+  featureIds: {value: Uint16Array([0, 1, 1, ...]), size: 1}
+}
+
+#### Array lengths
+
+Due to MultiPoints, the length of `properties` and `fields` arrays will not always be the same length (multiplied by `positions.size`) as the `positions` array. The length will match the count of individual features. This is in contrast to the `featureId` and `numericProp` arrays, which will contain the same number of elements as the `positions` array (divided by `positions.size`).
+
+
+#### Example comparison
+
+```js
+geojson = {
+  type: 'FeatureCollection',
+  features: [{
+    id: 123,
+    type: 'Feature',
+    properties: {name: 'London', population: 10000000},
+    geometry: {coordinates: [1.23, 4.56], type: 'Point'}
+  },
+  ...
+  ]
+}
+
+binary = {
+  points: {
+    positions: {value: Float32Array([1,23, 4.56, ...]), size: 2},
+    properties: [{name: 'London'}, ...],
+    numericProps: {
+      population: {value: Float32Array([10000000, ...], size: 1}
+    }
+    featureIds: {value: Uint16Array([0, ...]), size: 1}
+    fields: [{id: 123}]
+  }
+}
+```
+
+#### Lines
+
+Lines are represented in a similar manner, with the addition of a `pathIndices` array, which contains a series of offsets into the `positions` array, specifying where each line begins. All the other parameters are as above, namely that `featureIds` and `numericProps` are stored per-vertex, while `properties` and `fields` are per-feature.
+
+Here is how lines are represented, the first four vertices belong to the first line, thus the value of the second path index is 4.
+
+```js
+lines: {
+  positions: {value: Float32Array([x0, y0, ..., x4, y4, ...]), size: 2},
+  properties: [{name: 'name0'}, {name: 'name1}, ...],
+  numericProps: {
+    population: {value: Float32Array([100, 100, 100, 100, 789, 789, ...], size: 1}
+  }
+  pathIndices: {value: Uint16Array([0, 4, ...]), size: 1}
+  featureIds: {value: Uint16Array([0, 0, 0, 0, 1, 1, ...]), size: 1}
+  fields: [{id: 123}, {id: 456}]
+}
+```
+
+#### Polygons
+
+Polygons are an extension of the idea introduced with lines, but instead of `pathIndices` the `polygonIndicies` array specifies where each polygon starts inside the `positions` array. Because pygons can have holes, the offsets for the outer and inner rings are stored separately in the `primitivePolygonIndices` array. A polygon that has an outer ring consisting of 60 vertices and a hole with 40 vertices is represented as:
+
+```js
+polygons: {
+  positions: {value: Float32Array([x0, y0, ...]), size: 2},
+  polygonIndices: {value: Uint16Array([0, 100, ...]), size: 1}
+  primitivePolygonIndices: {value: Uint16Array([0, 60, 100, ...]), size: 1}
+}
+```
+
+#### Global feature ids
+
+Because the `features` array in the GeoJSON con contain a mix of different geometry types, in order to represent this ordering each of the `points`, `lines` and `polygons` objects contains a `globalFeatureIds` array, which contains the per-vertex indices into the original GeoJSON `features' array.
```

Comment:
```suggestion
Because the `features` array in the GeoJSON can contain a mix of different geometry types, in order to represent this ordering each of the `points`, `lines` and `polygons` objects contains a `globalFeatureIds` array, which contains the per-vertex indices into the original GeoJSON `features' array.
```